week-1-ts-ecosystem/w01e04-light-me-up/src/models/PowerSource.ts

```diff
@@ -0,0 +1,15 @@
+const DEFAULT_POWER_SUPPLY = 100;
+
+class PowerSource {
+  #energySupply = DEFAULT_POWER_SUPPLY;
+
+  public consume(energy: number): void {
+    if (this.#energySupply <= 0) {
+      throw Error('Source is out of power!');
+    }
+
+    this.#energySupply -= energy;
```

Comment:
Tutaj byłbym o tyle ostrożniejszy że możemy faktycznie wylądować z `energySupply` poniżej `0` a to "nielogiczne" zjawisko.
Więc drobna poprawka co do warunku.